feature_check.h

```diff
@@ -95,4 +95,18 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #endif  //(__cplusplus >= 202002L) || (defined(_MSVC_LANG) && _MSVC_LANG >=
         // 202002L)
 
+//
+// XYZ_HAS_STD_TYPE_IDENTITY
+// The macro is defined when std::type_identity_t<T> is available.
+//
+
+#ifdef XYZ_HAS_STD_TYPE_IDENTITY
+#error "XYZ_HAS_STD_TYPE_IDENTITY is already defined"
+#endif  // XYZ_HAS_STD_TYPE_IDENTITY
+
+#if (__cplusplus >= 202002L) || (defined(_MSVC_LANG) && _MSVC_LANG >= 202002L)
+#define XYZ_HAS_STD_TYPE_IDENTITY
+#endif  //(__cplusplus >= 202002L) || (defined(_MSVC_LANG) && _MSVC_LANG >=
+// 202002L)
```

Comment:
This macro is set but never used. You use `type_identity_t` below without testing this macro.

Reply:
It's used in tests to disable tests that won't run on older C++ versions.

Reply:
Yeah, but it's not used to guard those features in `indirect.h`. So for example this just errors out:
```
clang++ -O2 -std=c++17 -c indirect_test.cc
```
Theoretically you could support CTAD in C++17 (because CTAD itself was a C++17 feature); but right now you have only your C++14 version (which can't use CTAD) and your C++20 version (which assumes full C++20 support).

I'm figuring all this out as I write, so, forgive me if this is all old news. I guess the README already implies that C++17 isn't a supported target. You can probably at least partially ignore this comment, then.

I'm still confused why `XYZ_HAS_EXTENDED_CONSTRUCTOR_TEMPLATE_ARGUMENT_DEDUCTION` is handled/set/tested in a different way from `XYZ_HAS_TEMPLATE_ARGUMENT_DEDUCTION`, though.